pkg/git/git_test.go

```diff
@@ -64,6 +67,84 @@ func TestGetDefaultKubernetesRepoURLSuccess(t *testing.T) {
 	}
 }
 
+// createTestRepository creates a test repo, cd into it and returns the path
+func createTestRepository() (repoPath string, err error) {
```

Comment:
can `newSUT()` be reused here?

Reply:
I considered it, but I figured that instead of looking into only simulating the call by regenerating the fakes with counterfeiter (and possibly getting myself into a mess) it would be better to do a function that _actually_ calls git in a repo. Do you think I should go with the simulated tests? (can do, no problem :)

Reply:
Ok I've delved a bit into this. Currently, there is no way to use `newSUT()` as the function only creates a mock `Repository` interface for the tests and most of the calls to `git` call the command package directly.

I was thinking about how to add the command fakes to the git package following @saschagrunert 's advice and we would have to something like this:
1. Add a new interface to _interface_ (duh) with the command package. `CommandCaller` or something like that.
2. Replace all calls to `command.*` in the git package to go through this interface.
3. Create the mock CommandCaller in the tests and then rewrite the tests with it.

What do you think @hasheddan @saschagrunert  ? If this sounds good to you, I can rework the package later this week.